### PR TITLE
feat: externalIp support array and optimize api-server-external-service

### PR DESCRIPTION
**Pull request by qiuwei68**
#### What type of PR is this?
/kind feature

#### What does this PR do?

externalIp support array and optimize api-server-external-service
#### Which issue(s) does this PR fix?

Fixes #
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/kubenest/controlplane/endpoint.go [hunk 1]
```diff
@@ -53,6 +54,30 @@ func installApiServerExternalEndpointInVirtualCluster(dynamicClient dynamic.Inte
 			return errors.Wrap(err, "failed to convert kubernetes endpoint to typed object")
 		}
 
+		// Check if api-server-external-service endpoint already exists
+		_, err = dynamicClient.Resource(schema.GroupVersionResource{
+			Group:    "",
+			Version:  "v1",
+			Resource: "endpoints",
+		}).Namespace(constants.DefaultNs).Get(context.TODO(), constants.ApiServerExternalService, metav1.GetOptions{})
+		if err == nil {
```
Comment (village-way) [inline pkg/kubenest/controlplane/endpoint.go]: why not use `apierrors.IsNotFound(err)`

File: pkg/kubenest/init.go [hunk 2]
```diff
@@ -35,6 +35,7 @@ type initData struct {
 	virtualClusterDataDir string
 	privateRegistry       string
 	externalIP            string
+	externalIps           []string
```
Comment (village-way) [inline pkg/kubenest/init.go:38]: add `externalDonmains` ?

File: pkg/kubenest/controlplane/endpoint.go (hunk 1 above)
Comment (qiuwei68) [reply pkg/kubenest/controlplane/endpoint.go]: "err == nil" is used to check if any type of error has occurred, while "apierrors.IsNotFound(err)" is specifically used to check for the "resource not found" error. Therefore, the scope of "err == nil" is broader than that of "apierrors.IsNotFound(err)." In this case, if an error occurs, there could be multiple types of errors, not just the "resource not found" error.

File: pkg/kubenest/init.go (hunk 2 above)
Comment (qiuwei68) [reply pkg/kubenest/init.go:38]: yes, externalIp support add domain names, ipv4 and ipv6

**Pull request merged**